### PR TITLE
Modernize more tests, ref #3743

### PR DESCRIPTION
**Pull request by rzhao271**
Ref #3743

Changes made in this Pull Request:
 - Changed more tests to use `approx` and `assert_allclose` rather than `assert_almost_equals` and `assert_array_almost_equals`.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/core/test_atomgroup.py [hunk 1]
```diff
@@ -25,12 +25,7 @@
 from os import path
 
 import numpy as np
-
-from numpy.testing import (
-    assert_almost_equal,
-    assert_equal,
-    assert_array_almost_equal,
-)
+from numpy.testing import (assert_allclose, assert_equal)
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/core/test_atomgroup.py]: no longer need the brackets
```suggestion
from numpy.testing import assert_allclose, assert_equal
```

File: testsuite/MDAnalysisTests/analysis/test_align.py [hunk 2]
```diff
@@ -28,17 +28,12 @@
...
 from MDAnalysisTests.datafiles import (PSF, DCD, CRD, FASTA, ALIGN_BOUND,
                                        ALIGN_UNBOUND, PDB_helix)
-from numpy.testing import (
-    assert_almost_equal,
-    assert_equal,
-    assert_array_equal,
-    assert_array_almost_equal,
-    assert_allclose,
-)
+from numpy.testing import (assert_equal, assert_array_equal, assert_allclose)
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/analysis/test_align.py]:
```suggestion
from numpy.testing import assert_equal, assert_array_equal, assert_allclose
```

File: testsuite/MDAnalysisTests/formats/test_libmdaxdr.py [hunk 3]
```diff
@@ -24,8 +24,7 @@
 
 import numpy as np
 
-from numpy.testing import (assert_almost_equal, assert_array_almost_equal,
-                           assert_array_equal, assert_equal)
+from numpy.testing import (assert_allclose, assert_array_equal, assert_equal)
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/formats/test_libmdaxdr.py]:
```suggestion
from numpy.testing import assert_allclose, assert_array_equal, assert_equal
```

File: testsuite/MDAnalysisTests/lib/test_util.py [hunk 4]
```diff
@@ -31,9 +32,7 @@
 import copy
 
 import numpy as np
-from numpy.testing import (assert_equal, assert_almost_equal,
-                           assert_array_almost_equal, assert_array_equal,
-                           assert_allclose)
+from numpy.testing import (assert_equal, assert_array_equal, assert_allclose)
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/lib/test_util.py]:
```suggestion
from numpy.testing import assert_equal, assert_array_equal, assert_allclose
```

File: testsuite/MDAnalysisTests/utils/test_qcprot.py [hunk 5]
```diff
@@ -134,14 +135,15 @@ def test_CalcRMSDRotationalMatrix():
...
+           "fragments A and B does not match excpected value.")
 
     expected_rot = np.array([
         [0.72216358, -0.52038257, -0.45572112],
         [0.69118937, 0.51700833, 0.50493528],
         [-0.0271479, -0.67963547, 0.73304748]])
-    assert_almost_equal(rot.reshape((3, 3)), expected_rot, 6,
-                        "Rotation matrix for aliging B to A does not have expected values.")
+    assert_allclose(rot.reshape((3, 3)), expected_rot, rtol=0, atol=1e-6,
+                    err_msg="Rotation matrix for aliging B to A does not have expected values.")
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/utils/test_qcprot.py]: even though historical, let's fix the PEP8 issue here
```suggestion
    errmsg="Rotation matrix for aliging B to A does not have expected values."
    assert_allclose(rot.reshape((3, 3)), expected_rot, rtol=0, atol=1e-6,
                    err_msg=errmsg)
```

File: testsuite/MDAnalysisTests/utils/test_streamio.py [hunk 6]
```diff
@@ -371,19 +374,21 @@ def test_PDBQTReader(self, streamData):
...
         u = MDAnalysis.Universe(streamData.as_NamedStream('GRO'))
         assert_equal(u.atoms.n_atoms, 6)
-        assert_almost_equal(u.atoms[3].position,
-                            10. * np.array([1.275, 0.053, 0.622]), 3,  # manually convert nm -> A
-                            err_msg="wrong coordinates for water 2 OW")
-        assert_almost_equal(u.atoms[3].velocity,
-                            10. * np.array([0.2519, 0.3140, -0.1734]), 3,  # manually convert nm/ps -> A/ps
-                            err_msg="wrong velocity for water 2 OW")
+        assert_allclose(u.atoms[3].position,
+                        10. * np.array([1.275, 0.053, 0.622]), # manually convert nm -> A
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/utils/test_streamio.py]:
```suggestion
                        # manually convert nm -> A
                        10. * np.array([1.275, 0.053, 0.622]),
```

File: testsuite/MDAnalysisTests/utils/test_streamio.py [hunk 7]
```diff
@@ -371,19 +374,21 @@ def test_PDBQTReader(self, streamData):
...
+                        rtol=0, atol=1e-3,
+                        err_msg="wrong coordinates for water 2 OW")
+        assert_allclose(u.atoms[3].velocity,
+                        10. * np.array([0.2519, 0.3140, -0.1734]), # manually convert nm/ps -> A/ps
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/utils/test_streamio.py]:
```suggestion
                        # manually convert nm/ps -> A/ps
                        10. * np.array([0.2519, 0.3140, -0.1734]),
```

File: testsuite/MDAnalysisTests/utils/test_streamio.py [hunk 8]
```diff
@@ -395,5 +400,6 @@ def test_XYZReader(self, streamData):
         assert_equal(u.trajectory.frame, 1)  # !!!! ???
         u.trajectory.next()  # frame 2
         assert_equal(u.trajectory.frame, 2)
-        assert_almost_equal(u.atoms[2].position, np.array([0.45600, 18.48700, 16.26500]), 3,
-                            err_msg="wrong coordinates for atom CA at frame 2")
+        assert_allclose(u.atoms[2].position, np.array([0.45600, 18.48700, 16.26500]),
+                        rtol=0, atol=1e-3,
+                        err_msg="wrong coordinates for atom CA at frame 2")
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/utils/test_streamio.py]:
```suggestion
        assert_allclose(u.atoms[2].position,
                        np.array([0.45600, 18.48700, 16.26500]),
                        rtol=0, atol=1e-3,
                        err_msg="wrong coordinates for atom CA at frame 2")
```

File: testsuite/MDAnalysisTests/utils/test_streamio.py [hunk 9]
```diff
@@ -354,10 +355,12 @@ def test_PSF_CRD(self, streamData):
...
-        assert_almost_equal(u.atoms.total_charge(), self.ref_charmm_totalcharge, 3,
-                            "Total charge (in CHARMM) does not match expected value.")
-        assert_almost_equal(u.atoms.select_atoms('name H').charges, self.ref_charmm_Hcharges, 3,
-                            "Charges for H atoms do not match.")
+        approx_total_charge = approx(self.ref_charmm_totalcharge, abs=1e-3)
+        assert u.atoms.total_charge() == approx_total_charge, ("Total "
+               "charge in (CHARMM) does not match expected value.")
+        assert_allclose(u.atoms.select_atoms('name H').charges, self.ref_charmm_Hcharges,
+                        rtol=0, atol=1e-3,
+                        err_msg="Charges for H atoms do not match.")
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/utils/test_streamio.py]:
```suggestion
        assert_allclose(u.atoms.select_atoms('name H').charges,
                        self.ref_charmm_Hcharges,
                        rtol=0, atol=1e-3,
                        err_msg="Charges for H atoms do not match.")
```

File: testsuite/MDAnalysisTests/utils/test_streamio.py [hunk 10]
```diff
@@ -354,10 +355,12 @@ def test_PSF_CRD(self, streamData):
     def test_PQRReader(self, streamData):
         u = MDAnalysis.Universe(streamData.as_NamedStream('PQR'))
         assert_equal(u.atoms.n_atoms, self.ref_n_atoms)
-        assert_almost_equal(u.atoms.total_charge(), self.ref_charmm_totalcharge, 3,
-                            "Total charge (in CHARMM) does not match expected value.")
-        assert_almost_equal(u.atoms.select_atoms('name H').charges, self.ref_charmm_Hcharges, 3,
-                            "Charges for H atoms do not match.")
+        approx_total_charge = approx(self.ref_charmm_totalcharge, abs=1e-3)
+        assert u.atoms.total_charge() == approx_total_charge, ("Total "
+               "charge in (CHARMM) does not match expected value.")
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/utils/test_streamio.py]:
```suggestion
        errmsg = "Total charge in (CHARMM) does not match expected value."
        assert u.atoms.total_charge() == approx_total_charge, errmsg
```

**Review by IAlibay — CHANGES_REQUESTED** [review]
Couple things, mostly pep8 formatting

File: testsuite/MDAnalysisTests/lib/test_util.py [hunk 11]
```diff
@@ -437,23 +436,22 @@ def test_triclinic_box(self, lengths, angles):
                              comb_wr([-10, 0, 20, 70, 90, 120, 180], 3))
     def test_box_volume(self, lengths, angles):
         box = np.array(lengths + angles, dtype=np.float32)
-        assert_almost_equal(mdamath.box_volume(box),
-                            np.linalg.det(self.ref_trivecs(box)),
-                            decimal=5)
+        approx_det = approx(np.linalg.det(self.ref_trivecs(box)), abs=1e-5)
```
Comment (rzhao271) [inline testsuite/MDAnalysisTests/lib/test_util.py]: These tests are passing but it turns out one of the values wasn't a scalar. Will use `assert_allclose` instead.

**Review by rzhao271 — COMMENTED** [review]
Will also move more error messages to their own variables

**Comment by IAlibay** [comment]
> Will also move more error messages to their own variables

That's not a style requirement by the way, I suggested moving a few things because you had PEP8 violations, as long as pep8speaks is happy, I would vote for not changing things around.

**Comment by rzhao271** [comment]
Is this PR ready to merge?

**Comment by IAlibay** [comment]
> Is this PR ready to merge?

Sorry @rzhao271 the last week has been quite busy, I'll try to get to re-reviewing over the next couple of days.

**Comment by rzhao271** [comment]
Bump. Also let me know if you need me to change more test files before this PR can be merged.
Edit Dec 4: Closing this PR for now due to inactivity and also because I'm busier these days.

**Pull request closed without merging**